File: arrays/two_sums/two_sums.py
""" Solution Module """
class Solution:
    """ Solution Class """
    def run_solution(self, test_input):
        """ Solution Method """
        [nums, target] = test_input
        obj = {} # this will store the {diff: index}
        # Enumerate creates key/value for value and index [[1,2,5,3,7], 7]
        for index, value in enumerate(nums):
            diff = target - value
            # If the difference exists already in the hashMap we have both of our indices.
            if diff in obj:
                return [obj[diff], index]
            obj[value] = index

File: arrays/two_sums/two_sums.ts
export class Solution {
    /**
    * @param {string} testInput
    * @return {any} Returns either boolean to validate the test cases or a custom result. 
    */
    run_solution(testInput: any):[number, number] | undefined{
        const [nums, target] = testInput;
        const hashMap:Record<number,number> = {};
        let result 
        for(let idx=0; idx<nums.length; idx++) {
            const diff = target - nums[idx];
            const keys = Object.keys(hashMap)
            if(keys.includes(diff.toString())) {
                return [hashMap[diff], idx];
            }
            hashMap[nums[idx]] = idx;
        }
        return result
    }
}